packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/AppImpl.kt

```diff
@@ -65,18 +68,16 @@ public class AppImpl(
         // "isOnline" messages in short order. So in order to prevent resetting the network
         // too often we throttle messages, so a reconnect can only happen ever 5 seconds.
         RealmLog.debug("Network state change detected. ConnectionAvailable = $connectionAvailable")
-        val nowMs: Long = RealmInstant.now().let { timestamp ->
-            timestamp.epochSeconds * 1000L + timestamp.nanosecondsOfSecond / 1_000_000L
-        }
-        if (connectionAvailable && (lastOnlineStateReportedMs == null || nowMs - lastOnlineStateReportedMs!! > reconnectThresholdMs)
+        val now: Duration = RealmInstant.now().toDuration()
+        if (connectionAvailable && (lastOnlineStateReported == null || now.minus(lastOnlineStateReported!!) > reconnectThreshold)
```

Comment:
🎉  ... I guess you could make it more Kotlin idiomatic with something like
```
if (connectionAvailable && (lastOnlineStateReported?.let { now - it > reconnectThreshold } != false))
```
but don't mind keeping it as is.